pyiron/dft/job/generic.py

```diff
@@ -273,3 +267,42 @@ def set_empty_states(self, n_empty_states=None):
         raise NotImplementedError(
             "The set_empty_states function is not implemented for this code."
         )
+
+def get_k_mesh_by_cell(cell, kpoints_per_angstrom=1):
+    """
+    Args:
+        cell:
+        kpoints_per_angstrom:
+    Returns:
+    """
+    omega = np.linalg.det(cell)
+    l1, l2, l3 = cell
+    g1 = 2 * np.pi / omega * np.cross(l2, l3)
+    g2 = 2 * np.pi / omega * np.cross(l3, l1)
+    g3 = 2 * np.pi / omega * np.cross(l1, l2)
+
+    kmesh = np.rint(
+        np.array([np.linalg.norm(g) for g in [g1, g2, g3]]) * kpoints_per_angstrom
+    )
+    if kmesh.min() <= 0:
+        warnings.warn("kpoint per angstrom too low")
+    return [int(k) for k in kmesh]
+
+def get_k_mesh_by_density(cell, kmesh_density_per_inverse_angstrom=1.0):
```

Comment:
I'm not really sure if it's a good idea to offer both `get_k_mesh_by_cell` and `get_k_mesh_by_density`, because the difference is not very clear at first glance. I guess in the end it's easier for the people to remember that they have to put either a number or its inverse. Also I don't think it's rather unlikely that people need both of them.

Reply:
Agree, there is a bit of confusion.
I like @liamhuber  "kpoints_per_reciprocal_angstrom" notation.